src/gui/app/directives/modals/currency/addOrEditCurrency/addOrEditCurrency.js

```diff
@@ -56,6 +56,10 @@
                     return;
                 }
 
+                if (!$ctrl.currency.offline || $ctrl.currency.offline === "") {
+                    $ctrl.currency.offline = undefined;
```

Comment:
This is a great solution for currencies that are created or edited, but it will not fix the issue retroactively. So we'd need to find an additional solution without having to manually update all currencies. I'm currently not sure where though, possibly on load?

Reply:
I can do it on-load, no problem (in both [backend/currency/currency-access.ts](https://github.com/phroggster/Firebot/blob/602ddd0898023657ae440c3cb3d563f07e9fac01/src/backend/currency/currency-access.ts#L50), and [gui/app/services/currency.service.js](https://github.com/phroggster/Firebot/blob/602ddd0898023657ae440c3cb3d563f07e9fac01/src/gui/app/services/currency.service.js#L13)). The existing erroneous empty strings can just hang out in `currency.json` until the user edits any currency and the file gets rewritten. Commit coming soon, once it passes muster tests.

Thanks for spotting and calling out this oversight!

Reply:
I'm marking this resolved, but I'm not exceedingly happy about *any* of the decisions that I've made to correct this. Re-writing the JSON immediately after load from one location saves from patching it up in memory in the two places immediately after both loads, and should cover **all** bases. But now we've got a read-only caching layer making changes to a file.

I need a shower.